tests/scene/test_text_edit.h

```diff
@@ -7445,7 +7445,7 @@ TEST_CASE("[SceneTree][TextEdit] viewport") {
 	text_edit->set_line_as_last_visible(visible_lines * 2);
 	MessageQueue::get_singleton()->flush();
 	CHECK(text_edit->get_first_visible_line() == visible_lines);
-	CHECK(text_edit->get_v_scroll() == visible_lines);
+	CHECK(floor(text_edit->get_v_scroll()) == visible_lines);
```

Comment:
For most of these you can add the fractional part. It should be `_get_visible_lines_offset()`, but since it isn't exposed, something like:
```gdscript
	const float v_offset = 1.0 - ((text_edit->get_size().y - text_edit->get_minimum_size().y) / text_edit->get_line_height() - visible_lines);
```

Reply:
I have found that `(text_edit->get_v_scroll_bar()->get_max() - floor(text_edit->get_v_scroll_bar()->get_max()))` fixes it, also it's similar to the change that have caused the issue `set_v_scroll(get_scroll_pos_for_line(first_line, next_line.y) + (v_scroll->get_max() - floor(v_scroll->get_max())));` in `void TextEdit::set_line_as_last_visible(int p_line, int p_wrap_index)`.